action/gen.go

```diff
@@ -63,6 +104,21 @@ func (gen *GenOpts) LocalRun() error {
 		Data: eData,
 	})
 
+	// check for a metrics template and get its source
```

Comment:
I feel our design for reading in `*.metrics.tpl` and creating `*.metrics.yaml` in `gen`, and then reading `*.metrics.yaml` again within the task, is a bit hairy, and whole lot more complex than it needs to be.
***
Do you think it makes sense to simplify this logic by simply supplying a list of providers to the experiment task?

```yaml
- task: collect-metrics-database
  with:
    providers:
    - istio
    - knative
    versionInfo: ...
```

The above design will allow you to specifically read `istio.metrics.yaml`, and `knative.metrics.yaml` within the `collect-metrics-database` task.

You can factor out `providers` as a value, and supply a default value in the chart (`[istio]` is the default value for the istio-slo-validation chart of course).